src/main/java/com/google/vinet/data/MatchingRunner.java

```diff
@@ -0,0 +1,193 @@
+  private Set<VolunteerTimeSlot> volunteerTimeSlots;
+  private LocalDate date;
+
+  /**
+   * Construct a MatchingRunner with none of its data set.
+   */
+  public MatchingRunner() {
+
+  }
+
+  /**
+   * Construct a matching runner with a dependency on the provided DataStore implementation.
+   * @param datastore The DataStore implementation that this MatchingRunnner will depend on.
+   */
+  protected MatchingRunner(DatastoreService datastore) {
+    assert(datastore != null);
+    this.datastore = datastore;
+  }
+
+  /**
+   * Run the matching algorithm and store the results in DataStore.
+   * Any necessary data that is not already set will be pulled from DataStore.
+   */
+  public void run() {
+    if (this.datastore == null) this.datastore = DatastoreServiceFactory.getDatastoreService();
+    if (this.date == null) this.date = LocalDate.now().plusDays(1);
+    if (this.isolateTimeSlots == null) this.isolateTimeSlots = this.fetchIsolateTimeSlots();
+    if (this.volunteerTimeSlots == null) this.volunteerTimeSlots = this.fetchVolunteerTimeSlots();
+
+    deletePreviousMatches(this.date, this.datastore);
+
+    final Set<IsolateTimeSlot> matches = MatchingAlgorithm.matchTimeSlots(isolateTimeSlots, volunteerTimeSlots);
+
+    for (IsolateTimeSlot matching : matches) {
+      Entity matchingEntity = new Entity("Matching");
+      matchingEntity.setProperty("isolateId", matching.getIsolate().userId);
+      matchingEntity.setProperty("volunteerId", matching.getPairedSlot().registeredUser.userId);
+      matchingEntity.setProperty("date", matching.date.toString());
+      matchingEntity.setProperty("start", matching.start.toString());
+      matchingEntity.setProperty("end", matching.end.toString());
+      matchingEntity.setProperty("ticket", KeyFactory.keyToString(matching.ticket));
+      datastore.put(matchingEntity);
+    }
+  }
+
+  public static void deletePreviousMatches(LocalDate date, DatastoreService datastore){
+    Query query = new Query("Matching")
+                          .setFilter(new FilterPredicate("date", FilterOperator.LESS_THAN, date.toString()))
+                              .setKeysOnly();
+
+    PreparedQuery preparedQuery = datastore.prepare(query);
+
+    List<Entity> oldEntries =  preparedQuery.asList(FetchOptions.Builder.withDefaults());
+
+    for (Entity entry : oldEntries){
+      datastore.delete(entry.getKey());
+    }
+  }
+
+  /**
+   * @return the IsolateTimeSlots scheduled for this MatchingRunner's date from this MatchingRunner's implementation of DataStore.
+   */
+  public Set<IsolateTimeSlot> fetchIsolateTimeSlots() {
+    return fetchIsolateTimeSlots(this.date, this.datastore);
+  }
+
+  /**
+   * @return the VolunteerTimeSlots scheduled for this MatchingRunner's date from this MatchingRunner's implementation of DataStore.
+   */
+  public Set<VolunteerTimeSlot> fetchVolunteerTimeSlots() {
+    return fetchVolunteerTimeSlots(this.date, this.datastore);
+  }
```

Comment:
Are these delegate methods for internal use, or accessor methods for other classes?

Reply:
I've revisited the visibility and changed it to protected now https://github.com/googleinterns/step226-2020/pull/10/commits/1dae967203cd769af8d313997d59ceda518c51ba